### ISSUE TITLE
Mongock is not compatible with the native build

### ISSUE THREAD

**Issue by avdev4j** [jhipster/generator-jhipster-quarkus#179]
**Describe the bug**
Mongock is a library to help us to handle the data migration with a MongoDb database.
JHipster already uses it with Mongo and we reused the same with Quarkus.

Mongock uses `org.reflections` that cause an error when we build it with GraalVM to generate a native executable.

**To Reproduce**

Generate a project with MongoDb as a database.
Install GraalVM 
```
openjdk 11.0.10 2021-01-19
OpenJDK Runtime Environment GraalVM CE 21.0.0.2 (build 11.0.10+8-jvmci-21.0-b06)
OpenJDK 64-Bit Server VM GraalVM CE 21.0.0.2 (build 11.0.10+8-jvmci-21.0-b06, mixed mode, sharing)
```
Run the following command: `./mvnw clean verify -Pnative -DskipTests`

```
Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: Unsupported features in 2 methods
Detailed message:
Error: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved type during parsing: org.apache.commons.vfs2.VFS. To diagnose the issue you can use the --allow-incomplete-classpath option. The missing type is then reported at run time when it is accessed the first time.
Trace: 
	at parsing org.reflections.vfs.Vfs$DefaultUrlTypes$7.createDir(Vfs.java:287)
Call path from entry point to org.reflections.vfs.Vfs$DefaultUrlTypes$7.createDir(URL): 
	at org.reflections.vfs.Vfs$DefaultUrlTypes$7.createDir(Vfs.java:287)
	at org.reflections.vfs.Vfs.fromURL(Vfs.java:98)
	at org.reflections.vfs.Vfs.fromURL(Vfs.java:90)
	at org.reflections.Reflections.scan(Reflections.java:236)
	at org.reflections.Reflections$1.run(Reflections.java:199)
	at java.lang.Thread.run(Thread.java:834)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:519)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)
	at com.oracle.svm.core.code.IsolateEnterStub.PosixJavaThreads_pthreadStartRoutine_e1f4a8c0039f8337338252cd8734f63a79b5e3df(generated:0)
Error: com.oracle.graal.pointsto.constraints.UnresolvedElementException: Discovered unresolved type during parsing: org.apache.commons.vfs2.VFS. To diagnose the issue you can use the --allow-incomplete-classpath option. The missing type is then reported at run time when it is accessed the first time.
Trace: 
	at parsing org.reflections.vfs.Vfs$DefaultUrlTypes$7.matches(Vfs.java:281)
Call path from entry point to org.reflections.vfs.Vfs$DefaultUrlTypes$7.matches(URL): 
	at org.reflections.vfs.Vfs$DefaultUrlTypes$7.matches(Vfs.java:281)
	at org.reflections.vfs.Vfs.fromURL(Vfs.java:97)
	at org.reflections.vfs.Vfs.fromURL(Vfs.java:90)
	at org.reflections.Reflections.scan(Reflections.java:236)
	at org.reflections.Reflections$1.run(Reflections.java:199)
	at java.lang.Thread.run(Thread.java:834)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:519)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)
	at com.oracle.svm.core.code.IsolateEnterStub.PosixJavaThreads_pthreadStartRoutine_e1f4a8c0039f8337338252cd8734f63a79b5e3df(generated:0)

	at com.oracle.graal.pointsto.constraints.UnsupportedFeatures.report(UnsupportedFeatures.java:129)
	at com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:770)
	... 8 more


```

**Expected behavior**
The native build should succeed without errors.

**Suggestions**
Move from Mongock to [Liquibase with Mongo](https://docs.liquibase.com/workflows/database-setup-tutorials/mongodb.html).
Find a way to make Mongock compatible with the native build.

**Additional context**
Use `./mvnw dependency:tree -Dverbose` to check which library is using VFS.
Check this issue: https://github.com/quarkusio/quarkus/issues/9801

cc @danielpetisme

**Comment by avdev4j** [comment]
closed by #180

**Comment by dieppa** [comment]
I have just realised about this ticket and it seems that has been somehow sorted out.

However is there anything that we can provide natively from Mongock to help on this or improve.

Just let me know.
